### ISSUE TITLE
The text character '<' is not usable in custom product attribute name

### ISSUE THREAD

**Issue by Tommo1973** [woocommerce/woocommerce#17820]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [ x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Create custom attribute in product edit screen ( Name it anything)
2. Add 3 attribute values and include the less than symbol (<) in one of these. Tick visible on product page and use for variations.
3. On Variations tab select create variations from all attributes.
4. The attribute with the less than character cannot be used for a variation and just defaults to Any  

## Expected/actual behavior

When I follow those steps, I see 2 variations with attribute values assigned and the one with the '<' character does not assign.

I was expecting to see all three attribute values generate  variations

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: https://www.spectrographic.rt-website-design.co.uk
Site URL: https://www.spectrographic.rt-website-design.co.uk
WC Version: 3.2.4
Log Directory Writable: ✔
WP Version: 4.9
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: –
Language: en_GB

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 7.0.25
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.54.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.4
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 115.67MB
Database Data Size: 80.65MB
Database Index Size: 35.02MB
wp_woocommerce_sessions: Data: 0.06MB + Index: 0.00MB
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_order_itemmeta: Data: 0.05MB + Index: 0.03MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.01MB + Index: 0.01MB
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wpstg1_ac_abandoned_cart_history_lite: Data: 0.09MB + Index: 0.00MB
wpstg1_ac_email_templates_lite: Data: 0.02MB + Index: 0.00MB
wpstg1_ac_guest_abandoned_cart_history_lite: Data: 0.02MB + Index: 0.00MB
wpstg1_ac_sent_history_lite: Data: 0.02MB + Index: 0.00MB
wpstg1_commentmeta: Data: 0.02MB + Index: 0.03MB
wpstg1_comments: Data: 1.52MB + Index: 0.25MB
wpstg1_duplicator_pro_entities: Data: 0.00MB + Index: 0.00MB
wpstg1_duplicator_pro_packages: Data: 0.02MB + Index: 0.00MB
wpstg1_links: Data: 0.02MB + Index: 0.02MB
wpstg1_options: Data: 1.52MB + Index: 0.22MB
wpstg1_pmxi_files: Data: 0.02MB + Index: 0.00MB
wpstg1_pmxi_history: Data: 0.02MB + Index: 0.00MB
wpstg1_pmxi_imports: Data: 0.02MB + Index: 0.00MB
wpstg1_pmxi_posts: Data: 0.06MB + Index: 0.00MB
wpstg1_pmxi_templates: Data: 0.02MB + Index: 0.00MB
wpstg1_postmeta: Data: 11.52MB + Index: 8.03MB
wpstg1_posts: Data: 6.52MB + Index: 1.23MB
wpstg1_rg_form: Data: 0.02MB + Index: 0.00MB
wpstg1_rg_form_meta: Data: 0.05MB + Index: 0.00MB
wpstg1_rg_form_view: Data: 1.52MB + Index: 0.45MB
wpstg1_rg_incomplete_submissions: Data: 0.00MB + Index: 0.00MB
wpstg1_rg_lead: Data: 0.16MB + Index: 0.03MB
wpstg1_rg_lead_detail: Data: 0.38MB + Index: 0.63MB
wpstg1_rg_lead_detail_long: Data: 0.08MB + Index: 0.00MB
wpstg1_rg_lead_meta: Data: 0.02MB + Index: 0.05MB
wpstg1_rg_lead_notes: Data: 0.02MB + Index: 0.03MB
wpstg1_termmeta: Data: 0.06MB + Index: 0.04MB
wpstg1_terms: Data: 0.09MB + Index: 0.09MB
wpstg1_term_relationships: Data: 0.27MB + Index: 0.22MB
wpstg1_term_taxonomy: Data: 0.13MB + Index: 0.11MB
wpstg1_usermeta: Data: 1.52MB + Index: 0.67MB
wpstg1_users: Data: 0.11MB + Index: 0.08MB
wpstg1_wfBadLeechers: Data: 0.00MB + Index: 0.00MB
wpstg1_wfBlockedIPLog: Data: 0.06MB + Index: 0.05MB
wpstg1_wfBlocks: Data: 0.00MB + Index: 0.00MB
wpstg1_wfBlocksAdv: Data: 0.00MB + Index: 0.00MB
wpstg1_wfConfig: Data: 0.26MB + Index: 0.01MB
wpstg1_wfCrawlers: Data: 0.00MB + Index: 0.00MB
wpstg1_wfFileMods: Data: 0.88MB + Index: 0.20MB
wpstg1_wfHits: Data: 0.54MB + Index: 0.12MB
wpstg1_wfHoover: Data: 0.00MB + Index: 0.00MB
wpstg1_wfIssues: Data: 0.00MB + Index: 0.00MB
wpstg1_wfKnownFileList: Data: 0.44MB + Index: 0.06MB
wpstg1_wfLeechers: Data: 0.10MB + Index: 0.16MB
wpstg1_wfLockedOut: Data: 0.00MB + Index: 0.00MB
wpstg1_wfLocs: Data: 0.00MB + Index: 0.00MB
wpstg1_wfLogins: Data: 0.13MB + Index: 0.04MB
wpstg1_wfNet404s: Data: 0.00MB + Index: 0.00MB
wpstg1_wfNotifications: Data: 0.01MB + Index: 0.00MB
wpstg1_wfPendingIssues: Data: 0.00MB + Index: 0.00MB
wpstg1_wfReverseCache: Data: 0.00MB + Index: 0.00MB
wpstg1_wfScanners: Data: 0.01MB + Index: 0.01MB
wpstg1_wfSNIPCache: Data: 0.00MB + Index: 0.00MB
wpstg1_wfStatus: Data: 0.08MB + Index: 0.04MB
wpstg1_wfThrottleLog: Data: 0.00MB + Index: 0.00MB
wpstg1_wfVulnScanners: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wpstg1_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wpstg1_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_order_itemmeta: Data: 0.20MB + Index: 0.20MB
wpstg1_woocommerce_order_items: Data: 0.06MB + Index: 0.02MB
wpstg1_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_sessions: Data: 0.08MB + Index: 0.01MB
wpstg1_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wpstg1_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wpstg1_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.05MB
wpstg1_yoast_seo_links: Data: 0.00MB + Index: 0.00MB
wpstg1_yoast_seo_meta: Data: 0.00MB + Index: 0.00MB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB
wp_comments: Data: 0.02MB + Index: 0.09MB
wp_duplicator_pro_entities: Data: 0.01MB + Index: 0.00MB
wp_duplicator_pro_packages: Data: 0.00MB + Index: 0.00MB
wp_ewwwio_images: Data: 3.47MB + Index: 1.50MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_nextend2_image_storage: Data: 0.01MB + Index: 0.00MB
wp_nextend2_section_storage: Data: 0.00MB + Index: 0.01MB
wp_nextend2_smartslider3_generators: Data: 0.00MB + Index: 0.00MB
wp_nextend2_smartslider3_sliders: Data: 0.01MB + Index: 0.00MB
wp_nextend2_smartslider3_sliders_xref: Data: 0.00MB + Index: 0.00MB
wp_nextend2_smartslider3_slides: Data: 0.04MB + Index: 0.00MB
wp_options: Data: 5.52MB + Index: 0.23MB
wp_pmxi_files: Data: 0.02MB + Index: 0.00MB
wp_pmxi_history: Data: 0.02MB + Index: 0.00MB
wp_pmxi_imports: Data: 0.02MB + Index: 0.00MB
wp_pmxi_posts: Data: 0.06MB + Index: 0.00MB
wp_pmxi_templates: Data: 0.02MB + Index: 0.00MB
wp_postmeta: Data: 21.55MB + Index: 13.77MB
wp_posts: Data: 7.42MB + Index: 2.14MB
wp_rg_form: Data: 0.02MB + Index: 0.00MB
wp_rg_form_meta: Data: 0.05MB + Index: 0.00MB
wp_rg_form_view: Data: 1.52MB + Index: 0.45MB
wp_rg_incomplete_submissions: Data: 0.00MB + Index: 0.00MB
wp_rg_lead: Data: 0.16MB + Index: 0.03MB
wp_rg_lead_detail: Data: 0.39MB + Index: 0.64MB
wp_rg_lead_detail_long: Data: 0.08MB + Index: 0.00MB
wp_rg_lead_meta: Data: 0.02MB + Index: 0.05MB
wp_rg_lead_notes: Data: 0.02MB + Index: 0.03MB
wp_slim_events: Data: 0.34MB + Index: 0.25MB
wp_slim_events_archive: Data: 0.00MB + Index: 0.00MB
wp_slim_stats: Data: 4.59MB + Index: 0.37MB
wp_slim_stats_archive: Data: 0.00MB + Index: 0.00MB
wp_termmeta: Data: 0.06MB + Index: 0.04MB
wp_terms: Data: 0.09MB + Index: 0.09MB
wp_term_relationships: Data: 0.28MB + Index: 0.19MB
wp_term_taxonomy: Data: 0.13MB + Index: 0.11MB
wp_usermeta: Data: 1.52MB + Index: 0.67MB
wp_users: Data: 0.13MB + Index: 0.14MB
wp_wfBadLeechers: Data: 0.00MB + Index: 0.00MB
wp_wfBlockedCommentLog: Data: 0.00MB + Index: 0.00MB
wp_wfBlockedIPLog: Data: 0.00MB + Index: 0.00MB
wp_wfBlocks: Data: 0.00MB + Index: 0.00MB
wp_wfBlocksAdv: Data: 0.00MB + Index: 0.00MB
wp_wfConfig: Data: 0.83MB + Index: 0.01MB
wp_wfCrawlers: Data: 0.00MB + Index: 0.00MB
wp_wfFileMods: Data: 1.48MB + Index: 0.31MB
wp_wfHits: Data: 0.53MB + Index: 0.12MB
wp_wfHoover: Data: 0.00MB + Index: 0.00MB
wp_wfIssues: Data: 0.01MB + Index: 0.00MB
wp_wfKnownFileList: Data: 0.76MB + Index: 0.10MB
wp_wfLeechers: Data: 0.02MB + Index: 0.02MB
wp_wfLockedOut: Data: 0.00MB + Index: 0.00MB
wp_wfLocs: Data: 0.00MB + Index: 0.00MB
wp_wfLogins: Data: 0.05MB + Index: 0.02MB
wp_wfNet404s: Data: 0.00MB + Index: 0.00MB
wp_wfNotifications: Data: 0.02MB + Index: 0.00MB
wp_wfPendingIssues: Data: 0.00MB + Index: 0.00MB
wp_wfReverseCache: Data: 0.00MB + Index: 0.00MB
wp_wfScanners: Data: 0.00MB + Index: 0.00MB
wp_wfSNIPCache: Data: 0.00MB + Index: 0.00MB
wp_wfStatus: Data: 0.10MB + Index: 0.04MB
wp_wfThrottleLog: Data: 0.00MB + Index: 0.00MB
wp_wfVulnScanners: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_termmeta: Data: 0.00MB + Index: 0.00MB
wp_wpgmza: Data: 0.02MB + Index: 0.00MB
wp_wpgmza_categories: Data: 0.02MB + Index: 0.00MB
wp_wpgmza_category_maps: Data: 0.02MB + Index: 0.00MB
wp_wpgmza_maps: Data: 0.02MB + Index: 0.00MB
wp_wpgmza_polygon: Data: 0.02MB + Index: 0.00MB
wp_wpgmza_polylines: Data: 0.02MB + Index: 0.00MB
wp_yoast_seo_links: Data: 0.07MB + Index: 0.01MB
wp_yoast_seo_meta: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

acf: 4
attachment: 5343
customize_changeset: 1
global_product_addon: 1
ml-slide: 8
ml-slider: 1
nav_menu_item: 208
page: 100
post: 116
product: 492
product_tabpage: 2
product_variation: 702
revision: 5
shop_coupon: 1
shop_order: 5
slide: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (23) ###

Gravity Forms: by rocketgenius – 2.2.5
Advanced Custom Fields: Repeater Field: by Elliot Condon – 2.0.1
Advanced Custom Fields: by Elliot Condon – 4.4.12
Asset Queue Manager: by Nate Wright – 1.0.3
Childify Me: by Rocco Aliberti – 1.1.1
Display Posts Shortcode: by Bill Erickson – 2.9.0
Duplicator Pro: by Snap Creek – 3.5.3
Gravity Forms - Placeholders add-on: by Joan Piedra – 1.2.1
Max Mega Menu - StoreFront Integration: by Tom Hemsley – 1.0 – Not tested with the active version of WooCommerce
Max Mega Menu: by Tom Hemsley – 2.4
MinQueue: by Matthew Haines-Young – 1.1.2
Smart Slider 3 Pro: by Nextend – 3.2.8
Smart WooCommerce Search: by Stanislav S - YummyWP – 1.4.2 – Not tested with the active version of WooCommerce
Cookie Consent: by Catapult_Themes – 2.3.4
Woo Variations table: by Alaa Rihan – 1.3.10 – Not tested with the active version of WooCommerce
WooCommerce Purchase Order Payment Gateway: by WooCommerce – 1.1.5 – Not tested with the active version of WooCommerce
WooCommerce WorldPay Gateway: by Add On Enterprises – 3.3.6 – 3.6.1 is available – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.2.4
Wordfence Security: by Wordfence – 6.3.21
Yoast SEO: by Team Yoast – 5.8
WP Google Maps: by WP Google Maps – 6.4.06
WP-Optimize: by David Anderson
Ruhani Rabin
Team Updraft – 2.1.1

Slimstat Analytics: by Jason Crouse – 4.7.3.1

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #32 - /shop/
Basket: #33 - /cart/
Checkout: #34 - /checkout/
My account: #35 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront_Child
Version: 1.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Storefront
Parent Theme Version: 2.2.5
Parent Theme Author URL: https://woocommerce.com/
WooCommerce Support: ✔

### Templates ###

Overrides: storefront_child/woocommerce/emails/admin-new-order.php
storefront_child/woocommerce/emails/email-header.php
storefront_child/woocommerce/emails/email-order-details.php
storefront_child/woocommerce/emails/email-order-items.php
storefront_child/woocommerce/emails/email-styles.php
storefront_child/woocommerce/single-product/add-to-cart/simple.php

`
```
</details>

**Comment by mikejolley** [comment]
There is a fix here https://github.com/woocommerce/woocommerce/pull/17821/files

This will be included in 3.3 so there is longer to test - I don't think it should be included in a fix release yet.

Thanks

**Comment by Tommo1973** [comment]
@mikejolley, Many thanks for the quick fix.
